### PR TITLE
ICE Connection callback

### PR DESCRIPTION
**Pull request by dreamerns**
* Add support for monitoring ice connection change on local stream
* Add support to check if any stream got disconnected

### CODE REVIEW

File: src/stream.ts [hunk 1]
```diff
@@ -220,6 +231,8 @@ export class LocalStream extends Stream {
         delete this.mid;
       });
   }
+
```
Comment (jabuan) [inline src/stream.ts]: extra lines

**Review by jabuan — APPROVED** [review]
👍

File: src/client.ts [hunk 2]
```diff
@@ -167,6 +167,25 @@ export default class Client extends EventEmitter {
     this.dispatch.close();
   }
 
+  isStreamIceConnected = (local: boolean) => {
+    let connected = true;
+    const failedStatuses = ["closed", "disconnected", "failed"]
+    const streams = local ? this.localStreams as Stream[] : Object.values(this.streams) as Stream[];
```
Comment (timsu) [inline src/client.ts:173]: `Object.values` is a newer Javascript method and so the compiled output of ion-sdk-js won't work on older browsers like IE. This is why we have our own implementation in our codebase. I would avoid using it for compatibility reasons.

Comment (dreamerns) [reply src/client.ts:173]: Interesting, we do already have Object.values() in code. Its even in official repo https://github.com/pion/ion-sdk-js/blob/master/src/client.ts#L130

Comment (dreamerns) [reply src/client.ts:173]: In addition, IE doesn't support WebRTC anyway :-)

**Pull request merged**